botocore/utils.py

```diff
@@ -243,28 +238,108 @@ def set_value_from_jmespath(source, expression, value, is_first=True):
     source[current_key] = value
 
 
+class _RetriesExceededError(Exception):
+    """Internal exception used when the number of retries are exceeded."""
+    pass
+
+
 class InstanceMetadataFetcher(object):
 
-    _REQUIRED_CREDENTIAL_FIELDS = [
-        'AccessKeyId', 'SecretAccessKey', 'Token', 'Expiration'
-    ]
+    _RETRIES_EXCEEDED_ERROR = _RetriesExceededError
```

Comment:
We should rename this to `_RETRIES_EXCEEDED_ERROR_CLS` and raise instances of the class. Right now I think we are just raising the exception class.